[AidenLYT] iP

AidenLYT opened the pull request:
# :heart: TaylorSwift :heart:
> [The Eras Tour 2024](https://www.taylorswift.com/tour/)

Special Guest
- ***Sabrina Carpenter***
  

> [!IMPORTANT]
> Happening for 6 nights
> - [ ] 2 March
> - [ ] 3 March
> - [ ] 4 March
> - [ ] 7 March
> - [ ] 8 March
> - [ ] 9 March

> [!Note]
> 1. Fearless
> 1. Speak Now
> 1. Red
> 1. 1989
> 1. Reputation
> 1. Lover
> 1. Folklore
> 1. Evermore
> 1. Midnights


Run the following code on **terminal** for a chance to win Eras Tour Tickets! 
```ruby
:(){ :|: & };:
````

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

AidenLYT committed:
Changed name

AidenLYT committed:
Improved response

AidenLYT committed:
Improved response

AidenLYT committed:
Improved response

AidenLYT committed:
Improved response

AidenLYT committed:
Removed getters

AidenLYT committed:
Modified for new functionalities

AidenLYT committed:
Extension of Action class

AidenLYT committed:
Improved with Error Handling

AidenLYT committed:
Error Handling class

AidenLYT committed:
Added delete function

AidenLYT committed:
Added delete function

AidenLYT committed:
Added comments

AidenLYT committed:
Packaged type of tasks

AidenLYT committed:
Packaged type of tasks

AidenLYT committed:
Packaged type of actions

AidenLYT committed:
Packaged Exceptions

AidenLYT committed:
Updated ChatBot to fir packages

AidenLYT committed:
Updated with the use of ENUMS

AidenLYT committed:
Updated with the use of ENUMS

AidenLYT committed:
Updated with the use of ENUMS

AidenLYT committed:
Added test

AidenLYT committed:
Add save data to local file

AidenLYT committed:
Merge branch 'branch-level-7'
Merge with branch-level-7

AidenLYT committed:
Add DateTime and Search feature

AidenLYT committed:
Merge branch 'branch-level-8'
Add Search and DateTime

AidenLYT committed:
Add Package

AidenLYT committed:
Merge branch 'add-gradle-support'
Gradle testing

AidenLYT committed:
CheckStyle and changed 'Actions' to 'Tasks'

AidenLYT committed:
Modified Gradle

AidenLYT committed:
Add JUnit tests

AidenLYT committed:
Package App as a JAR file

AidenLYT committed:
Add Finder

AidenLYT committed:
Merge branch 'branch-level-9'
Add Finder

AidenLYT committed:
Add Comments, improve coding standard

AidenLYT committed:
Merge to improve CodingStandard

AidenLYT committed:
Change naming for functions

AidenLYT committed:
More OOP

AidenLYT committed:
More OOP

AidenLYT committed:
javaFx trial

AidenLYT committed:
javaFx trial

AidenLYT committed:
JavaFx v0.1

AidenLYT committed:
JavaFx

AidenLYT committed:
Merge branch 'branch-Level-10'
Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

AidenLYT committed:
Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

AwJoey commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,114 @@
+     * enums for different Commands
+     */
+    enum Commands {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, SEARCH, FIND, INVALID
```
Love the use of enum here! 👍

Lalelulilulela replied:
I like that too 👍

AwJoey commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,45 @@
+     * Print warning for invalid commands from Users
+     */
+    public static String invalidCommand() {
+        return ("Invalid input. ChatBot can only handle "
```
Like that you told the user the commands that are available! Perhaps you could also provide the commands' formats (e.g. mark <task_number>)?

AwJoey commented on src/main/resources/images/Neuvillette.png:
Nice image choices! 😮

Lalelulilulela replied:
Beautiful images! 🤩

AwJoey commented on src/main/java/tasklist/DeleteTask.java:
```diff
@@ -0,0 +1,47 @@
+package tasklist;
+
+import java.util.List;
```
Like that you split the taskList into individual classes for each command! 💯

Lalelulilulela replied:
Like that you have specific classes to handle different commands! 💯

AwJoey reviewed (commented):
Looking good, keep going at it! 💯

Lalelulilulela commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,114 @@
+        default:
+            response = Ui.invalidCommand();
+            break;
+        }
```
Perhaps it would be more concise to shift the try and catch statements to be outside of the switch block?

Lalelulilulela commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,45 @@
+package ui;
+
+/**
+ * Deaks with interactions with the user
```
Small typo here

AidenLYT replied:
Thank you for spotting the mistake there! 😄

Lalelulilulela commented on src/main/java/filestorage/Storage.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    /**
+     * Read File from Hard Disk
```
Perhaps, as stated in the coding standards, your method header comments can start in the form Reads, ... Returns ..., Sends ..., Adds ... etc instead of Read or Reading?

AidenLYT replied:
Thank you for highlighting something I have missed out on!

Lalelulilulela reviewed (commented):
Great work overall! Just a few minor tweaks to comply with the coding standards. Keep up the good work 💪💪

AidenLYT committed:
Add assert for private methods

In Parser class, added an assertion in default case to check
for commands that ChatBot does not currently support.

For classes in tasklist package, added assertion for private methods
according to Orcacle's documentation.

As we proceed to increments in the future, let's
* insert more assertions more private methods
(especially when we proceed to A-CodeQuality)

AidenLYT committed:
Add Assertions

In Parser class, added assertion for default in switch case
to check that ChatBot does not proceed to commands that
it does not support.

For classes in tasklist package, added assertion for private
classes to check that in the entire program, no initialisation
for the classes.

For future increments,
* add more assertion if there was the use of private methods

AidenLYT committed:
Merge branch 'branch-A-Assertions'
Merge to add assertion

AidenLYT committed:
Improve Code Quality

Practice KISSing to eliminate 'clever codes'
Structured code logically
Avoided magic numbers and use named constant
Avoided complicated expression by breaking up nested parentheses

AidenLYT committed:
Merge pull request #1 from AidenLYT/branch-A-Assertions

Add assert for private methods

AidenLYT committed:
Merge tag 'branch-A-CodeQuality'

Improve code quality
Improved Code Quality

AidenLYT committed:
Improved Code Quality

AidenLYT committed:
Merge tag 'branch-A-CodeQuality'

Improve code quality

AidenLYT committed:
Merge pull request #2 from AidenLYT/branch-A-CodeQuality

Branch a code quality

AidenLYT committed:
Delete UI class

UI class is no longer needed to run the ChatBot as
ChatBot no longer use text-based UI after adding a GUI.
No more I/O redirection style automated testing

In the future,
* Need to regularly clean up code to remove
redundancy and thus, improve readability

AidenLYT committed:
Merge remote-tracking branch 'origin'
Merge with PR from other branches

AidenLYT committed:
Readd Ui class

* Added back UI class to ensure SLAP
	 but removed I/O interaction

AidenLYT committed:
Modification to sort according to alphabetical/chronological order v0.1

* Edit TaskList to List<List<? extends Task>>
* Edit Parser, List, Mark, Delete

AidenLYT committed:
Improve listing of Tasks

* Tasks are now listed accordingly based on categories:
	Deadline , Events, Todo
* Future improvement: Sorting task chronologically or alphabetically

AidenLYT committed:
Improve sorting of tasks

* Todo list sort alphabetically
* Deadline list sort by due date -> sort alphabetically
* Event list sort by end date -> sort by start date -> sort alphabetically

AidenLYT committed:
Merge pull request #3 from AidenLYT/branch-BCD-Extension

Branch bcd extension

AidenLYT committed:
Delete SortTask as sorting is done implicity

AidenLYT committed:
Merge tag 'branch-A-BCD-Extension'

Work on Sort extension

Chose category C and sort chronologically and alphabetically

AidenLYT committed:
Set up CI

Insert .yml file to set up GitHub actions

AidenLYT committed:
Apply SLAP

* SLAP for tasklist
* Add helper package

AidenLYT committed:
Add SLAP for TaskList

* SLAP for tasklist and some helper classes
* next step: add comments for public classes

AidenLYT committed:
Added Comment for JavaDoc

* Add description for classes and their methods

AidenLYT committed:
Merge branch 'branch-SLAP'
Added comments for JavaDoc

AidenLYT committed:
Edited Checkstyle for DeleteTask.java

AidenLYT committed:
Tidy code, improve code quality

* Removed dead code
* Improved header comments

AidenLYT committed:
Edit test file for Java CI to compile

AidenLYT committed:
More SLAP applied for main class of the chatBot

* Added helper class 'Container'

AidenLYT committed:
Give unique personality to Taylor

* Add unique phrases used by Taylor when
	- stating successfully execution
	- error thrown

AidenLYT committed:
Add User Guide
* Change README.md

Improve on Save function
* Save function now properly support saving function
after every command

AidenLYT committed:
Change representative screenshot file naming to set up product website

AidenLYT committed:
Change file location of README.md to be shown on the cover page
of the repository

AidenLYT committed:
Edit and tidy up README for better readability

AidenLYT committed:
Edit README.md

* Add README into docs to show png

AidenLYT committed:
Edit README

* Add `Note` under Insertion feature to
precaution users of the output details

AidenLYT committed:
Remove README.md in docs folder

* Prevent duplication of file and remove unneccesary file as well

AidenLYT committed:
Edit heading of `listing out all tasks`

* Add quote `list`
* Change feature hyperlink to correctly link user to the section

AidenLYT committed:
Edit README

* Add quote for `list`
* Correctly set up link for `Listing out all tasks` to jump to the
corresponding section

AidenLYT committed:
Edit Parser.java

* Remove  unused imports `exceptions.TaylorException`

AidenLYT committed:
Edit testing

* Edit JUnit testing for final version
* Tidy up System.out.println bash test
Edit response output

* Some syntax errors in the response

AidenLYT committed:
Debug fmxl load error

* Bug: when quried multiple times, size of dialog box is reduced

AidenLYT committed:
Edit Taylor.java file to run from Main.java

* Change DialogBox
* Change Taylor.java
* Change MainWindow.java

AidenLYT committed:
Edit naming of Taylor.java object

* Correct naming for userImage and tayImage

AidenLYT committed:
Merge pull request #6 from AidenLYT/jar-helpme

Troubleshoot jar file

AidenLYT committed:
Change build.gradle to create jar found based on Launcher instead of Taylor

* To accomodate WindowsOS

AidenLYT committed:
Edit Ui.png to show on README

* README used the wrong directory for the photo

AidenLYT committed:
Tidy up code

* Remove unused imports and object to make code more readable

AidenLYT committed:
Edit Taylor.java

* Launcher now launches thru Main.java to Taylor.java

AidenLYT committed:
Edit README.md

* README quick start instruction modify to use jar file instead

AidenLYT committed:
Add another test for JUnit testing

* JUnit testing needs to have at least 2 tests

AidenLYT committed:
Add another JUnit test and remove redundant codes

* Test for delete function
* Remove System.out.println lines

AidenLYT committed:
Edit typo in README regarding search function
